lib/cubits/journal_create_cubit.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_form_builder/flutter_form_builder.dart';
+import 'package:star_book/cubits/cubit_state/cubit_state.dart';
+import 'package:star_book/domain/models/journal/journal.dart';
+import 'package:star_book/domain/repository/journal_repo.dart';
+import 'package:star_book/presentation/shared/form_models/jounral_form_model.dart';
+
+class JournalCreateCubit extends Cubit<CubitState<Journal>> {
+  final GlobalKey<FormBuilderState> formKey;
+  final JournalRepo journalRepo;
+
+  JournalCreateCubit({required this.journalRepo, required this.formKey})
+      : super(const InitialState());
+
+  Future<void> postJournal({String? id, DateTime? createdAt}) async {
```

Comment:
No need for this createdAt and even i don't think there is a need of id

Reply:
Now I think, we need createdAt because we have a date-picker

Reply:
we will change updated at, and that will be handled by backend. thanks for bringing this to attention.